saleor/graphql/order/mutations/orders.py

```diff
@@ -514,3 +515,32 @@ class Meta:
         model = models.Order
         permissions = (OrderPermissions.MANAGE_ORDERS,)
         public = False
+
+
+class RequestInvoice(BaseMutation):
+    class Arguments:
+        order_id = graphene.Int(
+            required=True, description="ID of the order related to invoice."
+        )
+        number = graphene.String(
+            required=False, description="Invoice number (optional)."
+        )
+
+    class Meta:
+        description = "Request an invoice for the order."
+        permissions = (OrderPermissions.MANAGE_ORDERS,)
+
+    @classmethod
+    def perform_mutation(cls, _root, info, **data):
+        try:
+            order = models.Order.objects.get(id=data.get("order_id"))
+        except models.Order.DoesNotExist:
+            raise ValidationError(
+                "Order with provided id does not exist.", code=OrderErrorCode.NOT_FOUND,
+            )
+
+        invoice = models.Invoice.objects.create(status=InvoiceStatus.PENDING)
+        info.context.extensions.invoice_request(
+            order=order, invoice=invoice, number=data.get("number")
+        )
+        return RequestInvoice()
```

Comment:
Don't we want to return something here?

Reply:
What would you suggest? after all this is a request of the invoice creation where the process starts. It has nothing to return, it either throws an error or it doesn't.

Reply:
Don't we want to use Job objects here? The idea behind them is to represent async actions like generating files etc.

Reply:
It can also be added as a separate PR.